filters/auth/grantconfig_test.go

```diff
@@ -0,0 +1,101 @@
+package auth_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zalando/skipper/filters/auth"
+	"github.com/zalando/skipper/secrets"
+)
+
+func TestOAuthConfigInit(t *testing.T) {
+	tempDir := t.TempDir()
+	missingIDFile := tempDir + "/missing-id"
+	missingSecretFile := tempDir + "/missing-secret"
+	existingFile := tempDir + "/existing"
+	require.NoError(t, os.WriteFile(existingFile, []byte("acontent"), 0644))
+
+	for i, tc := range []struct {
+		config        *auth.OAuthConfig
+		expectedError string
```

Comment:
this should be the error that we expect and not a string :)

Reply:
It's called from `skipper` package and lib users can test the error if they like. I think this should be good for all cases when we return an error. Much better then the `net` stdlib that you can test only by string compare.